clients/metadata/metadata_cassandra.go

```diff
@@ -1181,19 +1186,73 @@ const (
 		` WHERE ` + columnDestinationUUID + `=? and ` + columnName + `=?`
 )
 
-// CreateConsumerGroup creates a ConsumerGroup for the given destination, if it doesn't already exist
+func (s *CassandraMetadataService) CreateConsumerGroup(ctx thrift.Context, request *shared.CreateConsumerGroupRequest) (*shared.ConsumerGroupDescription, error) {
+	uuidRequest := shared.NewCreateConsumerGroupUUIDRequest()
+	uuidRequest.Request = request
+	uuidRequest.ConsumerGroupUUID = common.StringPtr(uuid.New())
+	return s.CreateConsumerGroupUUID(ctx, uuidRequest)
+}
+
+// CreateConsumerGroupUUID creates a ConsumerGroup for the given destination, if it doesn't already exist
 // ConsumerGroups are tied to a destination path, so the same ConsumerGroupName can be used across
 // multiple destination paths. If the requested [destinationPath, consumerGroupName] already exists,
 // this method will return an EntityAlreadyExistsError.
-func (s *CassandraMetadataService) CreateConsumerGroup(ctx thrift.Context, request *shared.CreateConsumerGroupRequest) (*shared.ConsumerGroupDescription, error) {
+func (s *CassandraMetadataService) CreateConsumerGroupUUID(ctx thrift.Context, request *shared.CreateConsumerGroupUUIDRequest) (*shared.ConsumerGroupDescription, error) {
+	createRequest := request.GetRequest()
+	cgUUID := request.GetConsumerGroupUUID()
+
+	// Dead Letter Queue destination creation
+
+	// Only non-UUID (non-DLQ) destinations get a DLQ for the corresponding consumer groups
+	// We may create a consumer group consume a DLQ destination and no DLQ destination creation needed in this case
+	var dlqUUID *string
```

Comment:
This method is too long now. Can you move the dlq creation logic into its method ?